src/routers/course.ts

```diff
@@ -32,41 +26,52 @@ courseRouter.post('/', async (req, res) => {
 	try {
 		const userID = req.header('userID');
 		if (userID) {
-			const course = new Course({
-				owner: userID,
-				name: req.body.name,
-			});
-			const created = await create(course);
-			res.json(created);
+			const body = req.body;
+			const checkPost = ajv.compile(
+				schema.components.schemas.CourseCreate
+			);
+
+			if (checkPost(body)) throw Error('body invalid');
+
+			const course = new Course({ ownerId: userID, ...body });
+
+			console.log(course); // TODO: call db operation
+			res.status(201); // + .json(created)
 		}
 	} catch (err) {
 		res.status(404).send('Not found.');
 	}
 });
 
-courseRouter.put('/', async (req, res) => {
+courseRouter.put('/:id', async (req, res) => {
 	try {
+		const id = req.params.id;
 		const userID = req.header('userID');
-		if (userID) {
-			const course = new Course({
-				id: req.body.id,
-				name: req.body.name,
-			});
-			const updated = await update(course, userID);
-			res.json(updated);
+		if (id && userID) {
+			const body = req.body;
+			const checkPut = ajv.compile(
+				schema.components.schemas.CourseUpdate
+			);
+
+			if (checkPut(body)) throw Error('body invalid');
+
+			const course = new Course(body);
```

Comment:
All good to skip the course here as well, and the same for other database operations in other routes